lib/src/embedded/js/executable.dart

```diff
@@ -0,0 +1,27 @@
+// Copyright 2024 Google Inc. Use of this source code is governed by an
+// MIT-style license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+import 'package:stream_channel/stream_channel.dart';
+
+import '../isolate_dispatcher.dart';
+import '../isolate_main.dart';
+import '../options.dart';
+import '../util/length_delimited_transformer.dart';
+import 'io.dart';
+import 'sync_receive_port.dart';
+import 'worker_threads.dart';
+
+void main(List<String> args) {
+  if (parseOptions(args)) {
+    if (isMainThread) {
+      IsolateDispatcher(StreamChannel.withGuarantees(stdin, stdout,
+                  allowSinkErrors: false)
+              .transform(lengthDelimited))
+          .listen();
+    } else {
+      var port = workerData! as MessagePort;
+      isolateMain(JSSyncReceivePort(port), JSSendPort(port));
```

Comment:
The distinction between this `isolateMain` and the one that's passed to `ReusableIsolate.spawn()` is very confusing. Given that _neither_ is consistently spawned as the main method of a new worker, it would be clearer to rename both to something that more explicitly describes what they do.

Reply:
I added a section and some diagrams in `lib/src/embedded/README.md` to explain how this convoluted process works.